File: dotnet/Mcma.Core/Model/ResourceEndpoint.cs
using System;

namespace Mcma.Core
{
    public class ResourceEndpoint : McmaObject
    {
        public string ResourceType { get; set; }

        public string HttpEndpoint { get; set; }

        public string AuthType { get; set; }

        public string AuthContext { get; set; }
    }
}

File: nodejs/base/core/lib/model/resource-endpoint.ts
import { McmaObject, McmaObjectProperties } from "./mcma-object";

export interface ResourceEndpointProperties extends McmaObjectProperties {
    resourceType: string;
    httpEndpoint: string;
    authType?: string;
    authContext?: any;
}

export class ResourceEndpoint extends McmaObject implements ResourceEndpointProperties {
    resourceType: string;
    httpEndpoint: string;
    authType?: string;
    authContext?: any;
    
    constructor(properties: ResourceEndpointProperties) {
        super("ResourceEndpoint", properties);

        this.checkProperty("resourceType", "string", true);
        this.checkProperty("httpEndpoint", "url", true);
        this.checkProperty("authType", "string", false);
    }
}
